hathor/builder/builder.py

```diff
@@ -239,30 +248,84 @@ def _get_or_create_pubsub(self) -> PubSubManager:
 
     def _get_or_create_rocksdb_storage(self) -> RocksDBStorage:
         assert self._rocksdb_path is not None
-        if self._rocksdb_storage is None:
-            self._rocksdb_storage = RocksDBStorage(path=self._rocksdb_path)
+
+        if self._rocksdb_storage is not None:
+            return self._rocksdb_storage
+
+        kwargs = {}
+        if self._rocksdb_cache_capacity is not None:
+            kwargs = dict(cache_capacity=self._rocksdb_cache_capacity)
+
+        self._rocksdb_storage = RocksDBStorage(
+            path=self._rocksdb_path,
+            **kwargs
+        )
+
         return self._rocksdb_storage
 
     def _get_or_create_tx_storage(self) -> TransactionStorage:
         if self._tx_storage is not None:
             return self._tx_storage
+
         if self._storage_type == StorageType.MEMORY:
             return TransactionMemoryStorage()
-        elif self._storage_type == StorageType.ROCKSDB:
+
+        if self._storage_type == StorageType.ROCKSDB:
             rocksdb_storage = self._get_or_create_rocksdb_storage()
             use_memory_index = self._force_memory_index
-            return TransactionRocksDBStorage(rocksdb_storage, use_memory_indexes=use_memory_index)
+
+            kwargs = {}
+            if self._rocksdb_with_index is not None:
+                kwargs = dict(with_index=self._rocksdb_with_index)
+
+            return TransactionRocksDBStorage(
+                rocksdb_storage,
+                use_memory_indexes=use_memory_index,
+                **kwargs
+            )
+
         raise NotImplementedError
 
+    def _get_or_create_event_storage(self) -> EventStorage:
+        if self._event_storage is not None:
+            pass
+        elif self._storage_type == StorageType.MEMORY:
+            self._event_storage = EventMemoryStorage()
+        elif self._storage_type == StorageType.ROCKSDB:
+            rocksdb_storage = self._get_or_create_rocksdb_storage()
+            self._event_storage = EventRocksDBStorage(rocksdb_storage)
+        else:
+            raise NotImplementedError
+
+        return self._event_storage
+
+    def _get_or_create_event_manager(self) -> Optional[EventManager]:
+        if self._event_manager is None and self._event_ws_factory is not None:
+            self._event_manager = EventManager(
+                reactor=self._get_reactor(),
+                pubsub=self._get_or_create_pubsub(),
+                event_storage=self._get_or_create_event_storage(),
+                event_ws_factory=self._event_ws_factory
+            )
+
+        return self._event_manager
```

Comment:
Shouldn't we assert that `self._event_ws_factory` is None?

Reply:
We could, but I left it like this in the same vein as letting the user set `use_memory()` and then `use_rocksdb()`. An user could use `enable_event_manager()` and then `set_event_manager()`.

Reply:
Got it. Thanks!